src/extension.ts

```diff
@@ -79,13 +79,20 @@ class VerySmartSelect {
         this.windowSelectionListener.dispose();
     }
 
-    private adjustSelection(range: Range) {
+    private updateSelections(selections: Selection[]) {
         const editor = window.activeTextEditor;
-        if (editor === undefined) {
-            return;
+        if (editor && selections.length > 0) {
+            this.didUpdateSelections = true;
+            editor.selections = selections;
+        }
+    }
+
+    private updateSelectionsHistory(selections: Selection[]) {
+        const lastSelections = this.selectionsHistory.length > 0
+            ? this.selectionsHistory[this.selectionsHistory.length - 1]
+            : undefined;
+        if (lastSelections === undefined || !areSelectionsEqual(lastSelections, selections)) {
```

Comment:
Only append to the history if the selections have changed. So we don't have to repeatedly shrink even if we haven't grown.